common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -179,12 +179,26 @@ public static StringSearch getStringSearch(
       final UTF8String left,
       final UTF8String right,
       final int collationId) {
+
+    if (collationId == UTF8_BINARY_LCASE_COLLATION_ID) {
+      return getStringSearch(left, right);
+    }
+
     String pattern = right.toString();
     CharacterIterator target = new StringCharacterIterator(left.toString());
     Collator collator = CollationFactory.fetchCollation(collationId).collator;
     return new StringSearch(pattern, target, (RuleBasedCollator) collator);
   }
 
+  private static StringSearch getStringSearch(
+      final UTF8String left,
+      final UTF8String right) {
```

Comment:
naming could be better here:
- final UTF8String patternUTF8String,
- final UTF8String targetUTF8String